core/services/gateway/handlers/functions/handler.functions.go

```diff
@@ -178,7 +179,11 @@ func (h *functionsHandler) HandleUserMessage(ctx context.Context, msg *api.Messa
 		return ErrRateLimited
 	}
 	if h.subscriptions != nil && h.minimumBalance != nil {
-		if balance, err := h.subscriptions.GetMaxUserBalance(sender); err != nil || balance.Cmp(h.minimumBalance.ToInt()) < 0 {
+		balance, err := h.subscriptions.GetMaxUserBalance(sender)
+		if balance == nil {
+			balance = big.NewInt(0)
+		}
+		if err != nil || balance.Cmp(h.minimumBalance.ToInt()) < 0 {
 			h.lggr.Debug("received a message from a user having insufficient balance", "sender", msg.Body.Sender, "balance", balance.String())
```

Comment:
How about we lggr.Debug() separately if err != nil (to log error details) but still return the same message to the user?